maven-wrapper/src/main/java/org/apache/maven/wrapper/Installer.java

```diff
@@ -87,6 +91,14 @@ public Path createDist( WrapperConfiguration configuration )
             downloaded = Files.exists( localZipFile );
         }
 
+        if ( verifyDistributionSha256Sum )
+        {
+            verifier.verify( localZipFile,
+                    "distributionSha256Sum",
+                    Verifier.SHA_256_ALGORITHM,
+                    configuration.getDistributionSha256Sum() );
+        }
+
```

Comment:
I would like to execute this block only when new file is downloaded, now we will calculate checksum on every Maven execute.